File: axm3720_lab01/axm3720_lab01.cpp
//Aditya Mishra
//1001663720
//MacOS

#include<dirent.h>
#include<string.h>
#include<stdio.h>

#include<sys/stat.h>
#include <unistd.h>
#include<iostream>

using namespace std;

float CurrentdirectorySize(char altogether[]) // creating a function to calculate the size of the directory.
{
    char whole[250];
    float totalSize=0;
    struct dirent *dt_p;
    struct stat totalFile;
   
  
    DIR *directory = opendir(altogether);
    if(directory == NULL)
    {
        printf("Not a directory"); // Display "Cannot open Directory" if this is not a directory
        return 0;
    }
    while ((dt_p=readdir(directory))!=NULL) // take data until the directory is empty
    {
        if(strcmp(dt_p->d_name,".")==0 || strcmp(dt_p->d_name,"..")==0)
            continue;

            strcpy(whole,altogether);
            strcat(whole,"/");
            strcat(whole,dt_p->d_name);
            stat(whole,&totalFile);
            if(S_ISREG(totalFile.st_mode)) //checks if it is a file by using recursive
            {
                
                totalSize+=totalFile.st_size; // adds the total size of the file
            }
            else //other than file i.e a directory
            {
                totalSize+=CurrentdirectorySize(whole); // adds the total size of the file
            }
    }
    return totalSize; // Returns the total value size of this directory
}
int main()
{
    float size = CurrentdirectorySize((char *)".");
    printf("\nTotal size of the directory: %0.0f bytes \n",size); //displays the size of the whole directory
}

File: axm3720_lab01/axm3720_lab01.py
# Aditya Mishra 
# 1001663720
# python3
# MACOS
import os #fetching the content of this directory.

def currentDirectorySize(totalFile): #creating a function to calculate the size of ghe directory
    totalSize = os.path.getsize(totalFile)
    for whole in os.listdir(totalFile): # using for loop in directory to find the total number of files
        altogether = os.path.join(totalFile, whole)
        if os.path.isdir(altogether): # using recursive function to check if it is directory
            totalSize += directorySize(altogether) # adding the directory size
        elif os.path.isfile(altogether): #using recursive function to check if it is file
            totalSize += os.path.getsize(altogether) # adding the file size
    return totalSize #will return the total size of the directory

print ("\n" + "Total size of the directory:"+ " " + str(currentDirectorySize("."))+ " " + "bytes"+"\n") # displaying the size of whole directory
